relay-auth/src/lib.rs

```diff
@@ -302,16 +311,16 @@ pub struct PublicKey {
 impl PublicKey {
     /// Verifies the signature and returns the embedded signature
     /// header.
-    pub fn verify_meta(&self, data: &[u8], sig: &str) -> Option<SignatureHeader> {
-        let mut iter = sig.splitn(2, '.');
+    pub fn verify_meta(&self, data: &[u8], sig: SignatureRef) -> Option<SignatureHeader> {
```

Comment:
```suggestion
    pub fn verify_meta(&self, data: &[u8], sig: SignatureRef<'_>) -> Option<SignatureHeader> {
```

We usually include the lifetime in the generics.